File: tree/traverse.c
#include<stdio.h>
#include<stdlib.h>
struct node
{
    int data;
    struct node *left;
    struct node * right;
};
struct node *newnode(int data)
{
    struct node *ptr=(struct node *)malloc(sizeof(struct node));
    ptr->data=data;
    ptr->left=NULL;
    ptr->right=NULL;
    return ptr;
};
void inorder(struct node *root)
{
    if(root)
    {
        inorder(root->left);
        printf("%d\t",root->data);
        inorder(root->right);
    }
}
void postorder(struct node *root)
{
    if(root)
    {
        postorder(root->left);
        postorder(root->right);
        printf("%d\t",root->data);

    }
}
void preorder(struct node *root)
{
    if(root)
    {
        printf("%d\t",root->data);
        preorder(root->left);
        preorder(root->right);
    }
}
int main()
{
    struct node *root=newnode(1);
    root->left=newnode(2);
    root->right=newnode(3);
    inorder(root);
    printf("\n");
    postorder(root);
    printf("\n");
    preorder(root);
    return 0;
}

File: tree/traverse.py
class Node:
    def __init__(self,key):
        self.left=None
        self.right=None
        self.val=key
def inorder(root):
    if root:
        inorder(root.left)
        print(root.val)
        inorder(root.right)
def postorder(root):
    if root:
        postorder(root.left)
        postorder(root.right)
        print(root.val)
def preorder(root):
    if root:
        print(root.val)
        preorder(root.left)
        preorder(root.right)

root=Node(1)
root.left=Node(2)
root.right=Node(3)
root.left.left=Node(4)
root.left.right=Node(5)
root.right.left=Node(6)
root.right.right=Node(7)
print("Inorder traversal")
inorder(root)
print("\n post order traversal")
postorder(root)
print("\n preorder order")
preorder(root)
